app/services/return-versions/setup/existing/generate-from-existing-requirements.service.js

```diff
@@ -20,13 +20,17 @@ const FetchExistingRequirementsService = require('./fetch-existing-requirements.
  *
  * @param {string} returnVersionId - The UUID of the selected return version to copy requirements from
  *
- * @returns {Promise<object[]>} an array of return requirements generated from the existing return version and ready to
- * be persisted to the setup session
+ * @returns {Promise<object>}  - return an array of return requirements generated from the existing return version and ready to
+ * be persisted to the setup session, if the return version has multiple uploads and if the return version is for quarterly returns
```

Comment:
Thank you for correcting the return type. But might I suggest ...

```suggestion
 * @returns {Promise<object>} the return version's return requirements transformed into an array of session return
 * requirements ready to be persisted to the setup session, plus whether the return version allows multiple uploads and
 * quarterly returns
```